Code/Framework/AzCore/Tests/AZStd/String.cpp

```diff
@@ -2537,7 +2537,7 @@ namespace Benchmark
     {
         AZStd::string test1{ "foo bar"};
         AZStd::string test2{ "bar foo" };
-        for (auto _ : state)
+        for ([[maybe_unused]] auto _ : state)
```

Comment:
Why doesn't this warn when using clang for other platforms?

Reply:
Hmm, I don't know. I have clang 13 on Windows. I made a sample file:
```cpp
#include <benchmark/benchmark.h>

class MyFixture : public benchmark::Fixture {};

BENCHMARK_F(MyFixture, BM_memcpy)(benchmark::State& state) {
  for (auto _ : state) {
    [[maybe_unused]] int x;
  }
}
```

Compiling with:
```
clang++ -isystem E:/ws/3rdParty/packages/googlebenchmark-1.5.0-rev2-windows/googlebenchmark/include -O3 -std=c++17 -Werror -Weverything -Wno-c++98-compat -Wno-global-constructors -Wunused-variable -Wno-weak-vtables foo.cpp -fsyntax-only
```

On Windows, this results in 2 warnings:
```
foo.cpp:7:1: error: identifier '_benchmark_2MyFixture_BM_memcpy_Benchmark' is reserved because it starts with '_' at global scope [-Werror,-Wreserved-identifier]
BENCHMARK_F(MyFixture, BM_memcpy)(benchmark::State& state) {
^
```
```
foo.cpp:8:13: error: unused variable '_' [-Werror,-Wunused-variable]
  for (auto _ : state)
            ^
```

The [same code on Linux with clang 13](https://godbolt.org/z/8YfjGvh3P) compiles with no warnings. It does seem a bit silly to have to mark all of these as `[[maybe_unused]]`...